source/sigv4.c

```diff
@@ -794,6 +895,336 @@ static SigV4Status_t getCredentialScope( SigV4Parameters_t * pSigV4Params,
+        const char * headerKey;
+        SigV4Status_t sigV4Status = SigV4Success;
+
+        assert( canonicalRequest != NULL );
+        assert( canonicalRequest->pBufCur != NULL );
+
+        pBufLoc = canonicalRequest->pBufCur;
+        buffRemaining = canonicalRequest->bufRemaining;
+
+        keyLen = canonicalRequest->pHeadersLoc[ headerIndex ].key.dataLen;
+
+        headerKey = canonicalRequest->pHeadersLoc[ headerIndex ].key.pData;
+
+        for( i = 0; i < keyLen; i++ )
+        {
+            if( !( flags & SIGV4_HTTP_PATH_IS_CANONICAL_FLAG ) )
+            {
+                if( !( isTrimmableSpace( headerKey, i, keyLen, trimKeyLen ) ) )
+                {
+                    if( buffRemaining < 1 )
+                    {
+                        sigV4Status = SigV4InsufficientMemory;
+                        break;
+                    }
+                    else
+                    {
+                        *pBufLoc = tolower( headerKey[ i ] );
+                        pBufLoc++;
+                        trimKeyLen++;
+                        buffRemaining -= 1;
+                    }
+                }
+            }
+            else
+            {
+                *pBufLoc = canonicalRequest->pHeadersLoc[ headerIndex ].key.pData[ i ];
+                pBufLoc++;
+                buffRemaining -= 1;
+            }
+        }
+
+        if( buffRemaining < 1 )
+        {
+            sigV4Status = SigV4InsufficientMemory;
+        }
+        else
+        {
+            *pBufLoc = ';';
+            pBufLoc++;
+            /* Calculate the remaining buffer. 1 is added for ';' character. */
+            buffRemaining -= 1;
+            canonicalRequest->pBufCur = pBufLoc;
+            canonicalRequest->bufRemaining = buffRemaining;
+        }
+
+        return sigV4Status;
+    }
+
+    static SigV4Status_t appendSignedHeaders( size_t headerCount,
+                                              uint32_t flags,
+                                              CanonicalContext_t * canonicalRequest )
+    {
+        size_t noOfHeaders = 0, keyLen = 0, i = 0;
+        SigV4Status_t sigV4Status = SigV4Success;
+
+        assert( canonicalRequest != NULL );
+        assert( canonicalRequest->pBufCur != NULL );
+
+        for( noOfHeaders = 0; noOfHeaders < headerCount; noOfHeaders++ )
+        {
+            assert( ( canonicalRequest->pHeadersLoc[ noOfHeaders ].key.pData ) != NULL );
+            sigV4Status = writeSignedHeaderToCanonicalRequest( noOfHeaders, flags, canonicalRequest );
+
+            if( sigV4Status != SigV4Success )
+            {
+                break;
+            }
+        }
+
+        if( sigV4Status == SigV4Success )
+        {
+            /* Replacing the last ';' with '\n' as last header does need to have ';'. */
+            canonicalRequest->pBufCur--;
+            *canonicalRequest->pBufCur = '\n';
+            canonicalRequest->pBufCur++;
+        }
+
+        return sigV4Status;
+    }
```

Comment:
Please use `/*-----------------------------------------------------------*/` to separate each function implementation
```suggestion
    }

/*-----------------------------------------------------------*/
```